contracts/on-chain-claim/src/contract.rs

```diff
@@ -0,0 +1,141 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+pub use address_info::AddressInfo;
+multiversx_sc::imports!();
+multiversx_sc::derive_imports!();
+
+pub mod address_info;
+pub mod config;
+
+use crate::config::{MAX_REPAIR_GAP, SFT_AMOUNT};
+use multiversx_sc_modules::only_admin;
+
+#[multiversx_sc::contract]
+pub trait OnChainClaimContract: config::ConfigModule + only_admin::OnlyAdminModule {
+    #[init]
+    fn init(&self, repair_streak_token_id: TokenIdentifier) {
+        require!(
+            repair_streak_token_id.is_valid_esdt_identifier(),
+            "Invalid token ID"
+        );
+        self.repair_streak_token_identifier()
+            .set(repair_streak_token_id);
+        let caller = self.blockchain().get_caller();
+        self.add_admin(caller);
+    }
+
+    #[upgrade]
+    fn upgrade(&self) {}
+
+    #[endpoint(claim)]
+    fn claim(&self) {
+        let caller = self.blockchain().get_caller();
+        require!(
+            !self.blockchain().is_smart_contract(&caller),
+            "Only user accounts can perform claim"
+        );
+        self.require_same_shard(&caller);
+
+        let current_epoch = self.blockchain().get_block_epoch();
+
+        let address_info_mapper = self.address_info(&caller);
+        if address_info_mapper.is_empty() {
+            let address_info = AddressInfo::new(1, current_epoch, 1, 1);
```

Comment:
I'd make this a method of AddressInfo, maybe simply a `new_with_epoch` method where it has a single argument, of current_epoch, while filling the rest with a default value, i.e. 1.

Reply:
fixed